src/components/ProjectDetail/ProjectTasks.tsx

```diff
@@ -1,12 +1,92 @@
-import React from 'react';
+import { TASKS } from 'fakeData/tasks';
+import React, { useMemo } from 'react';
+import { DragDropContext } from 'react-beautiful-dnd';
+import { Task, TaskStatus } from 'utils/types';
+import StatusColumn from './Components/StatusColumn';
 import './styles/Tasks.scss';
 
 interface TasksProps {
    projectId: number;
 }
+interface ListTasks {
+   [key: string]: Task[];
+}
+const categories = [
+   TaskStatus.Requesting,
+   TaskStatus.Unscheduled,
+   TaskStatus.Doing,
+   TaskStatus.Reviewing,
+   TaskStatus.Completed,
+];
 
 function ProjectTasks({ projectId }: TasksProps) {
-   return <div className="projectdetail__task">ProjectTask</div>;
+   let listTasks = useMemo<ListTasks>(() => {
+      let lists: ListTasks = {};
+      const tasks = TASKS.filter((task) => task.projectId === projectId);
+      lists[TaskStatus.Requesting] = tasks.filter(
+         (task) => task.status === TaskStatus.Requesting
+      );
+      lists[TaskStatus.Unscheduled] = tasks.filter(
+         (task) => task.status === TaskStatus.Unscheduled
+      );
+      lists[TaskStatus.Doing] = tasks.filter(
+         (task) => task.status === TaskStatus.Doing
+      );
+      lists[TaskStatus.Reviewing] = tasks.filter(
+         (task) => task.status === TaskStatus.Reviewing
+      );
+      lists[TaskStatus.Completed] = tasks.filter(
+         (task) => task.status === TaskStatus.Completed
+      );
+      return lists;
+   }, [projectId]);
```

Comment:
```suggestion
   const [tasksMap, setTasksMap] = useState<TasksMap>({})
   useEffect(() => {
      let lists: TasksMap = {};
      const tasks = TASKS.filter((task) => task.projectId === projectId);
      lists[TaskStatus.Requesting] = tasks.filter(
         (task) => task.status === TaskStatus.Requesting
      );
      lists[TaskStatus.Unscheduled] = tasks.filter(
         (task) => task.status === TaskStatus.Unscheduled
      );
      lists[TaskStatus.Doing] = tasks.filter(
         (task) => task.status === TaskStatus.Doing
      );
      lists[TaskStatus.Reviewing] = tasks.filter(
         (task) => task.status === TaskStatus.Reviewing
      );
      lists[TaskStatus.Completed] = tasks.filter(
         (task) => task.status === TaskStatus.Completed
      );
      setTasksMap(lists);
   }, []);
```

listTasks ở đây có thể thay đổi nên e dùng useState nhé, useMemo dùng để tính toán 1 biến dựa trên các biến khác, k được thay đổi nó kiểu gán mới